pkg/controller/eventqueue.go

```diff
@@ -19,23 +20,23 @@ import (
 // for each events.
 type EventQueue struct {
 	queue              workqueue.RateLimitingInterface
-	process            func(interface{}) (bool, error)
+	process            func(queuedEvent) error
```

Comment:
While golang allow these function pointers to be passed around, would be a lot more comfortable if we had declared this as an `Interface`. Something on the lines of :

```
type EventProcessor interface {
Process(queuedEvent) error
}
```
and make the `controller` implement this interface and pass it as a field within `EventQueue`. Makes the object semantics a lot more cleaner. Currently a reader would need to look at references of the `NewEventQueue` method to understand that the controller somehow is also implements the `EventProcessor` loop.

Reply:
Absolutely agree with you.
Goal here was to make small, palatable changes for quick reviews. (Thank you!)
Will merge this and make a new PR to implement your proposal, which I love!